File: core/core-domain/src/main/java/com/coffee/gu/order/OrderMenus.java
package com.coffee.gu.order;

import com.coffee.gu.menu.Menu;

import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

public class OrderMenus {
    private final Map<Long, Menu> menuMap;

    private OrderMenus(Map<Long, Menu> menuMap) {
        this.menuMap = menuMap;
    }

    public static OrderMenus from(List<Menu> menus) {
        return new OrderMenus(menus.stream()
                .collect(Collectors.toMap(Menu::getId, menu -> menu)));
    }

    public Menu getByMenuId(Long menuId) {
        return menuMap.get(menuId);
    }

    public boolean matches(Set<Long> menuIds) {
        return menuMap.keySet().equals(menuIds);
    }
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/order/OrderMenus.kt
package com.coffee.gu.order

import com.coffee.gu.menu.Menu

class OrderMenus(
    val menuMap: Map<Long, Menu>,
) {
    fun getByMenuId(menuId: Long): Menu? {
        return menuMap[menuId]
    }

    fun matches(menuIds: Set<Long>): Boolean {
        return menuMap.keys == menuIds
    }

    companion object {
        fun from(menus: List<Menu>): OrderMenus {
            return OrderMenus(menus.associateBy { it.id })
        }
    }
}